src/Microsoft.AspNetCore.SignalR.Core/DefaultHubLifetimeManager.cs

```diff
@@ -161,8 +163,10 @@ public override Task SendGroupAsync(string groupName, string methodName, object[
             var group = _groups[groupName];
             if (group != null)
             {
-                var message = CreateInvocationMessage(methodName, args);
```

Comment:
Since you decided to touch this, we should fix it to do what SendToAll is doing (not allocating until there's something that isn't complete).

Reply:
I've refactored so all are improved. They avoid allocating message and `List<Task>` until necessary, and avoids multiple `List<Task>` when sending to multiple groups.